### PR TITLE
Cover scenario when connectionType is none for kubelogin in HelmDeployV0

### PR DESCRIPTION
**Pull request by qianz2**
**Task name**: HelmDeployV0

**Description**: Fixing issue https://github.com/microsoft/azure-pipelines-tasks/issues/17958

**Documentation changes required:** (Y/N) <Please mark if documentation changes are required>

**Added unit tests:** (Y/N) <Please mark if unit tests were added or updated according changes>

**Attached related issue:** (Y/N) <Please add link to related issue here>

**Checklist**:
- [ ] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [ ] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/HelmDeployV0/src/helm.ts [hunk 1]
```diff
@@ -78,10 +76,11 @@ function runHelmSaveCommand(helmCli: helmcli, kubectlCli: kubernetescli, failOnS
 
 async function run() {
     var command = tl.getInput("command", true).toLowerCase();
+    var connectionType = tl.getInput("connectionType", true);
     var isKubConfigRequired = isKubConfigSetupRequired(command);
     var kubectlCli: kubernetescli;
     if (isKubConfigRequired) {
-        var kubeconfigfilePath = command === "logout" ? tl.getVariable("KUBECONFIG") : await getKubeConfigFile();
+        var kubeconfigfilePath = (command === "logout" || connectionType === "None") ? tl.getVariable("KUBECONFIG") : await getKubeConfigFile();
```
Comment (qianz2) [inline Tasks/HelmDeployV0/src/helm.ts:83]: When connectionType is none and customers are using external auth like kubelogin, a kubeconfig file path would be required to initiate the kubernetesCli here. 
A build test passed with setting KUBECONFIG=~/kube/config in my local build agent.
https://mwittest.visualstudio.com/merlynop/_build/results?buildId=570&view=logs&j=275f1d19-1bd8-5591-b06b-07d489ea915a

**Comment by qianz2** [comment]
@rvairavelu @manolerazvan Could you help review this PR? Thanks

File: Tasks/HelmDeployV0/src/helm.ts [hunk 2]
```diff
@@ -78,17 +78,18 @@ function runHelmSaveCommand(helmCli: helmcli, kubectlCli: kubernetescli, failOnS
 
 async function run() {
     var command = tl.getInput("command", true).toLowerCase();
+    var connectionType = tl.getInput("connectionType", true);
     var isKubConfigRequired = isKubConfigSetupRequired(command);
     var kubectlCli: kubernetescli;
-    if (isKubConfigRequired) {
-        var kubeconfigfilePath = command === "logout" ? tl.getVariable("KUBECONFIG") : await getKubeConfigFile();
+    var externalAuth = connectionType === "None" && tl.getVariable("KUBECONFIG");
```
Comment (merlynomsft) [inline Tasks/HelmDeployV0/src/helm.ts:84]: I'm thinking it's better to set externalAuth only if connectionType === "None" and command is "install" or "upgrade". If "KUBECONFIG" is not set, we can log an error directly indicting that KUBECONFIG must be set when command is "install" or "upgrade". That would be more dev-friendly and avoid calling kubectlCli.login() when not needed

File: Tasks/HelmDeployV0/src/helm.ts [hunk 3]
```diff
@@ -181,7 +182,7 @@ function runHelm(helmCli: helmcli, command: string, kubectlCli: kubernetescli, f
             let manifests = helmutil.getManifestsFromRelease(helmCli, releaseName);
             if (manifests && manifests.length > 0) {
                 const manifestUrls = getManifestFileUrlsFromHelmOutput(output);
-                const allPods = JSON.parse(kubectlCli.getAllPods().stdout);
+                const allPods = JSON.parse((kubectlCli.getAllPods()).stdout);
```
Comment (merlynomsft) [inline Tasks/HelmDeployV0/src/helm.ts:185]: extra parenthesis not needed

File: Tasks/HelmDeployV0/src/helm.ts (hunk 2 above)
Comment (qianz2) [reply Tasks/HelmDeployV0/src/helm.ts]: sounds good, updated

File: Tasks/HelmDeployV0/src/helm.ts [hunk 4]
```diff
@@ -81,9 +81,12 @@ async function run() {
     var connectionType = tl.getInput("connectionType", true);
     var isKubConfigRequired = isKubConfigSetupRequired(command);
     var kubectlCli: kubernetescli;
-    var externalAuth = connectionType === "None" && tl.getVariable("KUBECONFIG");
+    var externalAuth = connectionType === "None" && (command === "install" || command === "upgrade");
+    if (externalAuth && !tl.getVariable("KUBECONFIG")) {
+        tl.error("KUBECONFIG kube configuration file path must be set when command is install or upgrade.");
```
Comment (merlynomsft) [inline Tasks/HelmDeployV0/src/helm.ts:86]: missing when "Connection Type" is none

"...when Connection Type is none and command is install or upgrade"

Comment (qianz2) [reply Tasks/HelmDeployV0/src/helm.ts]: updated

**Review by merlynomsft — APPROVED** [review]

**Pull request merged**